api/tests/Unit/UserAuthorizationScopeTest.php

```diff
@@ -0,0 +1,263 @@
+    {
+        parent::setUp();
+
+        $this->seed(RolePermissionSeeder::class);
+
+        $this->platformAdmin = User::factory()
+            ->asAdmin()
+            ->create();
+
+        $this->pool1 = Pool::factory()
+            ->for($this->platformAdmin)
+            ->published()
+            ->afterCreating(function (Pool $p) {
+                // normally this happens only when the process operator is added - might want to reconsider that logic
+                $p->team()->create(['name' => 'pool-'.$p->id]);
+            })
+            ->create();
+
+        $this->pool2 = Pool::factory()
+            ->for($this->platformAdmin)
+            ->published()
+            ->afterCreating(function (Pool $p) {
+                $p->team()->create(['name' => 'pool-'.$p->id]);
+            })
+            ->create();
+
+        $this->user1 = User::factory()
+            ->asApplicant()
+            ->create();
+
+        $this->user2 = User::factory()
+            ->asApplicant()
+            ->create();
+
+        $this->candidate1 = PoolCandidate::factory()
+            ->for($this->user1)
+            ->for($this->pool1)
+            ->create([
+                'submitted_at' => Carbon::now(),
+            ]);
+
+        $this->candidate2 = PoolCandidate::factory()
+            ->for($this->user2)
+            ->for($this->pool2)
+            ->create([
+                'submitted_at' => Carbon::now(),
+            ]);
+    }
+
+    // no tests for scopeAuthorizedToViewSpecific since it is never directly used by the graphql schema
+
+    // a guest should be able to view no users
+    public function testViewAsGuest(): void
+    {
+        Auth::shouldReceive('user')
+            ->andReturn(null);
+
+        $userIds = User::authorizedToView()->get()->pluck('id');
+        assertEqualsCanonicalizing([], $userIds->toArray());
+    }
+
+    // an applicant should be able to view just themselves
+    public function testViewAsApplicant(): void
+    {
+        Auth::shouldReceive('user')
+            ->andReturn($this->user1);
+
+        $userIds = User::authorizedToView()->get()->pluck('id');
+        assertEqualsCanonicalizing([
+            $this->user1->id,
+        ], $userIds->toArray());
+    }
+
+    // an pool operator should be able to view themselves and any users with submitted candidates in their pools
+    public function testViewAsPoolOperator(): void
+    {
+        $poolOperator = User::factory()
+            ->asPoolOperator($this->pool1->team->name)
+            ->create();
+        Auth::shouldReceive('user')
+            ->andReturn($poolOperator);
+
+        // can also see the user with an application submitted to team 1's pool
+        $userIds = User::authorizedToView()->get()->pluck('id');
+        assertEqualsCanonicalizing([
+            $poolOperator->id,
+            $this->user1->id,
+        ], $userIds->toArray());
+    }
+
+    // a request responder should be able to view any user
```

Comment:
Technically, this goes against the spec.

Reply:
This one I'll scope out into another ticket.